lib/src/eager_span_scanner.dart

```diff
@@ -84,6 +84,13 @@ class EagerSpanScanner extends SpanScanner {
     return character;
   }
 
+  @override
+  int readCodePoint() {
+    final codePoint = super.readCodePoint();
+    _adjustLineAndColumn(codePoint);
```

Comment:
Should  this override be removed too? If it calls the `super.readCodePoint`, and that calls `readChar` virtually, it also updates the line/column during those calls.

And add tests to figure out whether it works or not.